jkq/ddsim/pathqasmsimulator.py

```diff
@@ -0,0 +1,375 @@
+        inds = df['tensors'][i][1]
+        tags = df['tensors'][i][0]
+        tensors.append(qtn.Tensor(data, inds, tags, left_inds=inds[:len(inds) // 2]))
+    return tensors
+
+
+def create_tensor_network(qc):
+    import quimb.tensor as qtn
+    import sparse
+    from jkq.ddsim import dump_tensor_network
+
+    if isinstance(qc, QuantumCircuit):
+        filename = qc.name + '_' + str(qc.num_qubits) + ".tensor"
+        nqubits = qc.num_qubits
+    else:
+        filename = 'tensor.tensor'
+        nqubits = qc.header.n_qubits
+
+    dump_tensor_network(qc, filename)
+    tensors = read_tensor_network_file(filename)
+    os.remove(filename)
+
+    # add the zero state tensor |0...0> at the beginning
+    shape = [2] * nqubits
+    data = sparse.COO(coords=[[0]] * nqubits, data=1, shape=shape, sorted=True, has_duplicates=False)
+    inds = ['q' + str(i) + '_0' for i in range(nqubits)]
+    tags = ['Q' + str(i) for i in range(nqubits)]
+    tensors.insert(0, qtn.Tensor(data=data, inds=inds, tags=tags))
+
+    # using the following lines instead would allow much greater flexibility,
+    # but is not supported for DD-based simulation at the moment
+
+    # add the zero state tensor |0>...|0> at the beginning
+    # shape = [2]
+    # data = np.zeros(2)
+    # data[0] = 1.0
+    # for i in range(qc.num_qubits):
+    #     inds = ['q'+str(i)+'_0']
+    #     tags = ['Q'+str(i)]
+    #     tensors.insert(0, qtn.Tensor(data=data.reshape(shape), inds=inds, tags=tags))
+
+    return qtn.TensorNetwork(tensors)
+
+
+def get_contraction_path(qc, max_time: int = 60, max_repeats: int = 1024, dump_path: bool = True,
+                         plot_ring: bool = False):
+    import cotengra as ctg
+    from opt_einsum.paths import linear_to_ssa
+
+    tn = create_tensor_network(qc)
+
+    opt = ctg.HyperOptimizer(max_time=max_time,
+                             max_repeats=max_repeats,
+                             progbar=True,
+                             parallel=4,
+                             minimize='flops')
+    info = tn.contract(all, get='path-info', optimize=opt)
+    path = linear_to_ssa(info.path)
+
+    if dump_path:
+        if isinstance(qc, QuantumCircuit):
+            filename = qc.name + '_' + str(qc.num_qubits) + ".path"
+        else:
+            filename = 'contraction.path'
+        with open(filename, 'w') as file:
+            file.write(str(path))
+
+    if plot_ring:
+        fig = opt.get_tree().plot_ring(return_fig=True)
+        fig.savefig("contraction_ring.svg", bbox_inches='tight')
+
+    return path
+
+
+class PathQasmSimulator(BackendV1):
+    """Python interface to JKQ DDSIM Task-Based Simulator"""
+
+    SHOW_STATE_VECTOR = False
+
+    @classmethod
+    def _default_options(cls) -> Options:
+        return Options(
+            shots=None,
+            parameter_binds=None,
+            simulator_seed=None,
+            mode="sequential",
+            nGatesQC1=0,
+            nGatesQC2=0,
```

Comment:
typically in Python names are written in snake case (with underscores between words) instead of camel case.
-> `n_gates_qc1`

Reply:
moved from python to c++

Reply:
if these are moved to C++, are they needed at all?
what about the other configuration options? shouldn't bracket size and start point be here?